src/Common/src/System/Security/Cryptography/Asn1V2.cs

```diff
@@ -0,0 +1,5246 @@
+                
+                if (tag.TagClass != TagClass.Universal)
+                {
+                    throw new CryptographicException(SR.Cryptography_Der_Invalid_Encoding);
+                }
+
+                totalRead += headerLength;
+
+                if (isIndefinite && tag.TagValue == (int)UniversalTagNumber.EndOfContents)
+                {
+                    ValidateEndOfContents(tag, length, headerLength);
+
+                    bytesRead = totalRead;
+                    return totalContent;
+                }
+
+                if (tag.TagValue != (int)UniversalTagNumber.OctetString)
+                {
+                    // T-REC-X.690-201508 sec 8.7.3.2 (in particular, Note 2)
+                    throw new CryptographicException(SR.Cryptography_Der_Invalid_Encoding);
+                }
+
+                if (ruleSet == AsnEncodingRules.CER)
+                {
+                    if (tag.IsConstructed || lastSegmentLength != MaxCERSegmentSize)
+                    {
+                        // T-REC-X.690-201508 sec 9.2
+                        throw new CryptographicException(SR.Cryptography_Der_Invalid_Encoding);
+                    }
+                }
+
+                cur = cur.Slice(headerLength);
+
+                if (length == null)
+                {
+                    totalContent += CopyConstructedOctetString(
+                        cur,
+                        ref destination,
+                        write,
+                        ruleSet,
+                        true,
+                        ref lastSegmentLength,
+                        out int nestedBytesRead);
+
+                    totalRead += nestedBytesRead;
+                    cur = cur.Slice(nestedBytesRead);
+                }
+                else if (tag.IsConstructed)
+                {
+                    totalContent += CopyConstructedOctetString(
+                        Slice(cur, 0, length.Value),
+                        ref destination,
+                        write,
+                        ruleSet,
+                        false,
+                        ref lastSegmentLength,
+                        out int nestedContentRead);
+
+                    totalRead += nestedContentRead;
+                    cur = cur.Slice(nestedContentRead);
+                }
+                else
+                {
+                    int lengthValue = length.Value;
+
+                    // T-REC-X.690-201508 sec 9.2
+                    if (ruleSet == AsnEncodingRules.CER && lengthValue > MaxCERSegmentSize)
+                    {
+                        throw new CryptographicException(SR.Cryptography_Der_Invalid_Encoding);
+                    }
+
+                    totalRead += lengthValue;
+                    totalContent += lengthValue;
+                    lastSegmentLength = lengthValue;
+
+                    ReadOnlyMemory<byte> segment = Slice(cur, 0, lengthValue);
+             
```

Comment:
Perhaps also add an overload that accepts `BitArray`? Would be useful when >64 possible flags. (low-priority)

Possible implementations:
```csharp
        private void ConvertNamedBitList(Asn1Tag tag, ulong value)
        {
            // WARNING: COMPLETELY UNTESTED

            Span<byte> temp = stackalloc byte[sizeof(ulong)];
            temp.Clear(); // since stackalloc doesn't guarantee zero-init
            int indexOfHighestSetBit = -1;

            for (int i = 0; value != 0; value >>= 1, i++)
            {
                if ((value & 1) != 0)
                {
                    temp[i / 8] |= 0x80 >> (i % 8);
                    indexOfHighestSetBit = i;
                }
            }
            
            if (indexOfHighestSetBit < 0)
            {
                // No bits were set; this is an empty bit string.
                WriteBitString(tag, ReadOnlySpan<byte>.Empty);
            }
            else
            {
                // At least one bit was set; determine the shortest length necessary to represent the bit string.
                WriteBitString(
                    tag,
                    new ReadOnlySpan<byte>(temp, 0, (indexOfHighestSetBit / 8) + 1),
                    7 - (indexOfHighestSetBit % 8));
            }
        }

        private void ConvertNamedBitList(Asn1Tag tag, System.Collections.BitArray value)
        {
            // WARNING: COMPLETELY UNTESTED

            byte[] temp = new byte[value.Length / 8]; // use ArrayPool instead
            Array.Clear(temp, 0, temp.Length); // if ArrayPool doesn't guarantee zero-init
            int indexOfHighestSetBit = -1;

            for (int i = 0; i < value.Count; i++)
            {
                if (value[i])
                {
                    temp[i / 8] |= 0x80 >> (i % 8);
                    indexOfHighestSetBit = i;
                }
            }

            if (indexOfHighestSetBit < 0)
            {
                // No bits were set; this is an empty bit string.
                WriteBitString(tag, ReadOnlySpan<byte>.Empty);
            }
            else
            {
                // At least one bit was set; determine the shortest length necessary to represent the bit string.
                WriteBitString(
                    tag,
                    new ReadOnlySpan<byte>(temp, 0, (indexOfHighestSetBit / 8) + 1),
                    7 - (indexOfHighestSetBit % 8));
            }

            // release temp back to ArrayPool
        }
```
Also removes all implementations other than `NamedZeroIsOne`.